include/boost/math/fft/real_algorithms.hpp

```diff
@@ -346,6 +346,147 @@
       real_inverse_dft_prime_bruteForce_outofplace(in_first,in_last,out,sign);
   }
   
+  template <class T, class allocator_t>
+  void real_dft_composite_outofplace(
+            const T *in_first, 
+            const T *in_last, 
+            T* out, 
+            int sign,
+            const allocator_t& alloc)
+  {
+    /*
+      Cooley-Tukey mapping, intrinsically out-of-place, Decimation in Time
+      composite sizes.
+    */
+    using allocator_type = allocator_t;
+    using ComplexType = ::boost::multiprecision::complex<T>; 
```

Comment:
Notice here that the real-to-halfcomplex routine for composite sizes is an adaptation of the complex Cooley-Tukey FFT. In the algorithm we take advantage of the the complex arithmetic, hence we need to build a complex type from the real `T` for inner calculations only.

Reply:
Good note. Thanks for pointing this out.